src/components/ReminderCtaButton.tsx

```diff
@@ -1,123 +1,106 @@
 import React, { useState } from 'react';
-import { css, ThemeProvider } from '@emotion/react';
-import { neutral, brandAlt, news, body, space } from '@guardian/source-foundations';
+import { css, SerializedStyles, ThemeProvider } from '@emotion/react';
+import { body, space } from '@guardian/source-foundations';
 import { Button } from '@guardian/source-react-components';
 
-import { buildReminderFields, createReminder, ReminderStage } from '../logic/reminders';
-import type { InteractiveButtonStatus } from '../logic/types';
+import {
+    buildReminderFields,
+    createReminder,
+    ReminderStage,
+    ReminderComponent,
+} from '../logic/reminders';
+import type { InteractiveButtonStatus, Extras } from '../logic/types';
 
 import { FetchEmail } from '../types/dcrTypes';
 import { TrackClick } from '../utils/tracking';
-import { LoadingDots } from './CtaLoadingDotsAnimation';
+import { buildLoadingDots } from './CtaLoadingDotsAnimation';
+import { contributionsTheme, getColors, ColorStylesData } from '../styles/colorData';
 
-// Custom theme for Button/LinkButton
-// See also `tertiaryButtonOverrides` below.
-const buttonStyles = {
-    textPrimary: neutral[7],
-    backgroundPrimary: brandAlt[400],
-    backgroundPrimaryHover: brandAlt[300],
-    textSecondary: neutral[7],
-    backgroundSecondary: neutral[93],
-    backgroundSecondaryHover: neutral[86],
-    borderSecondary: neutral[86],
+const defaultButtonColors: ColorStylesData = {
+    styleReminderButton: '#121212',
+    styleReminderButtonBackground: '#f6f6f6;',
+    styleReminderButtonHover: '#dcdcdc',
 };
 
-const contributionsTheme = {
-    button: buttonStyles,
-    link: buttonStyles,
-};
-
-const styles = {
-    buttonWrapperStyles: css`
-        margin: ${space[4]}px ${space[2]}px ${space[1]}px 0;
-        display: flex;
-        flex-direction: column;
-        flex-wrap: wrap;
-        justify-content: flex-start;
-        align-items: flex-start;
-        max-width: 50%;
-    `,
-    buttonMargins: css`
-        margin: ${space[1]}px ${space[2]}px ${space[1]}px 0;
-    `,
-    button: css`
-        background-color: ${news[400]};
-        color: ${neutral[100]};
-        &:hover {
-            background-color: ${news[400]};
-        }
-    `,
-    thankYouText: css`
-        ${body.medium({ fontWeight: 'bold' })};
-        color: ${neutral[0]};
-        margin-top: ${space[3]}px;
-    `,
-    remindMeButtonOverrides: css`
-        background-color: transparent !important;
-        color: ${neutral[7]} !important;
-
-        :hover {
-            background-color: ${neutral[86]} !important;
-        }
-    `,
-    smallPrint: css`
-        color: ${neutral[0]};
-        margin-top: ${space[2]}px;
-        ${body.small()};
-    `,
+const getButtonStyles = (userVals: Extras, defaults: ColorStylesData) => {
+    const styles = getColors(userVals as Extras, defaults);
```

Comment:
Currently the return value of `getColours` is `ColorStylesData` and all of the fields in that type are optional - but we interpolate them into the CSS styles below as though they are guaranteed to be present. I think returning a type where the fields are guaranteed to be there would be a win in terms of type safety (as in the comment above). What do you reckon? Happy to chat this over/pair if it's useful!

Reply:
As mentioned above, will give it a go

Reply:
See above